src/wpool_time_checker.erl

```diff
@@ -84,26 +84,37 @@ handle_call({add_handler, Handler}, _, State = #state{handlers = Handlers}) ->
 %%%===================================================================
 %% @private
 -spec handle_info(any(), state()) -> {noreply, state()}.
-handle_info({check, Pid, TaskId, Runtime}, State) ->
+handle_info({check, Pid, TaskId, Runtime, WarningsLeft}, State) ->
   case erlang:process_info(Pid, dictionary) of
     {dictionary, Values} ->
       run_task(
         TaskId, proplists:get_value(wpool_task, Values), Pid,
-        State#state.wpool, State#state.handlers, Runtime);
+        State#state.wpool, State#state.handlers, Runtime, WarningsLeft);
     _ -> ok
   end,
   {noreply, State};
 handle_info(_Info, State) -> {noreply, State}.
 
-run_task(TaskId, {TaskId, _, Task}, Pid, Pool, Handlers, Runtime) ->
-  Args = [ {alert, overrun}, {pool, Pool}, {worker, Pid}, {task, Task}
-         , {runtime, Runtime}],
-  _ = [catch Mod:Fun(Args) || {Mod, Fun} <- Handlers],
+run_task(TaskId, {TaskId, _, Task}, Pid, Pool, Handlers, Runtime, 1) ->
```

Comment:
Shouldn't this be `0` instead of one?
If `max_overrun_warnings` is `1`, I would expect wpool to warn me once, then kill it.

Reply:
And that is just what it is happening here. If we use `0` then we'd have two warning messages and then it would be killed.

To be clear about this specific semantic: if the `overrun_warning` is 1000 and the `max_overrun_warnings` is `1` it would be killed after 1 sec. If the `max_overrun_warnings` is ' 2'  it would be killed after 3 secs (1000 + 2*1000 ms) and so on.

Reply:
I would go with your logic as long as we somehow indicate that the last warning is _not like the others_… it's not just a process that overrun, this time we're killing it.

Reply:
Besides, when you write the docs, don't forget to explain that the rounds of warnings are not equally timed… we do have an exponential backoff algorithm in place.

Reply:
What do you think about removing the exponential backoff when `max_overrun_warnings /= infinity` so that the effects of the configuration parameters are more _intuitive_?

Reply:
I like the linear exponential backoff on this case. It'd need to be clearly explained on the docs as well.

Reply:
OK. Go for it, then!

Reply:
Also, what do you think about only allow the `max_overrun_warnings /= infinity` when the strategy is `available_worker`? Maybe I'm wrong but on other strategies it seems that would cause some scheduled tasks to me missed

Reply:
Since strategies are interchangeable (i.e. you can run a call with `available_worker` followed by one with `random_worker` against the same pool), it makes no sense to restrict it. But I would add a warning in the docs.